### DISCUSSION TITLE
sgp4_tsince

### DISCUSSION THREAD

**Discussion by sxd190026** [brandon-rhodes/python-sgp4#100] [Q&A] [answered]
Hi, I want to use 'sgp4_tsince' function for orbit propagation. 

`satellite.sgp4_tsince(t)`
I don't know if the input variable 't' is time in seconds or minutes There's no description of it in documentation, so if anybody can help me that would be great.

The 'satellite' object is created from a TLE using 'Satrec' function.

**Comment by brandon-rhodes** [comment, marked as answer]
The time `t` is measured in minutes before (negative) or after (positive) the exact time and date of the satellite’s epoch time.

What has led you to abandon the documented `.sgp4()` method? We should update the documentation for your use case, since other people might find themselves in the same situation!

**Reply by sxd190026** [reply]
Thank you for your reply. Is there any way I can propagate a tle and retrieve the final orbital elements (e.g. mean motion, eccentricity, arg. of perigee etc.) using this package?

**Reply by brandon-rhodes** [reply]
Yes! Check out the "Attributes" section of the documentation to learn about the derived orbital elements.

And, again, it would be helpful to hear more about your use case, to understand why `sgp4_tsince()` rather than `sgp4()` will be your entry point, as that could help improve the documentation. Thanks!

**Reply by sxd190026** [reply]
I wanted to get the position of a satellite at every five seconds starting from a tle epoch. For that purpose, I created an array of time elapsed since tle epoch (in minutes) and used a for loop for calculating position and velocity at each instant of time provided in the array. Below is the code I used:

```
line1 = Data[2*idx[idx0]] #first line of tle
line2 = Data[2*idx[idx0] + 1] #second line of tle
satellite = sgp4.api.Satrec.twoline2rv(line1, line2)
DT = (epochs[idx[idx0]+1] - epochs[idx[idx0]]).seconds #time difference between one tle epoch and the next tle epoch in seconds
T = np.arange(5,DT,5)  #create the array fo time elapsed
T = np.append(T,DT)
T = [t/60 for t in T]; #convert from second to minute

r = [];v=[];
for t in T:
    e,r0,v0 = satellite.sgp4_tsince(t)
    r.append(r0)
    v.append(v0)
```

I checked the documentation for attributes. It has the orbital elements at the tle epoch, but I am looking for the final orbital elements after propagating a tle for a certain period of time. For example, I want to take one tle and propagate it until I have a second tle. I want to derive the orbital elements at the epoch of 2nd tle and compare it with elements provided in the 2nd tle.

**Reply by brandon-rhodes** [reply]
> I wanted to get the position of a satellite at every five seconds starting from a tle epoch. For that purpose, I created an array of time elapsed since tle epoch (in minutes) and used a for loop for calculating position and velocity at each instant of time provided in the array.

Thanks for providing sample code! Because `sgp4_tsince()` does seem to be easier for your use case than converting to Julian dates, I might try to better document that routine before the next release.

> I checked the documentation for attributes. It has the orbital elements at the tle epoch, but I am looking for the final orbital elements after propagating a tle for a certain period of time. For example, I want to take one tle and propagate it until I have a second tle. I want to derive the orbital elements at the epoch of 2nd tle and compare it with elements provided in the 2nd tle.

The attributes should be listed in the documentation’s "Attributes" section — search the web page for the phrase "singly averaged mean elements" and see what comes up.

**Reply by sxd190026** [reply]
Thanks a lot for your help. I was trying to manually input the parameters and create a 'Satrec' object using the following code:
```
from sgp4.api import Satrec
from sgp4.api import WGS84,WGS72
sat = Satrec()
satellite = sat.sgp4init(WGS84,'i',5,18441.785,2.8098e-05,6.969196665e-13,0.0,0.1859667,5.7904160274885,0.5980929187319,0.3373093125574,0.0472294454407,6.0863854713832)
print(satellite)
satellite1 = Satrec(WGS84,'i',5,18441.785,2.8098e-05,6.969196665e-13,0.0,0.1859667,5.7904160274885,0.5980929187319,0.3373093125574,0.0472294454407,6.0863854713832)
print(satellite1)
print(satellite1.am)
print(satellite1.ecco)
print(satellite1.argpo)
```

the output is:

```
None
<sgp4.wrapper.Satrec object at 0x0000021A15ECF7D0>
0.0
0.0
0.0
```

I have used the same parameters as given in the example in documentation. The 'satellite' object is registered as 'nonetype', therefore doesn't have any attributes. If I use 'Satrec' with the same parameters, it generates a Satrec object, however all its parameters are zero (I have shown some of them). propagating this  object doesn't generate any co-ordinates. Is the 'sgp4init' function deprecated? Let me know if there is any other way to do this or if there's something wrong on my part.

**Reply by brandon-rhodes** [reply]
The routine `sat.sgp4init()` rewrites the orbital parameters of `sat` itself, so it has no return value; it doesn’t create a new satellite. So simply keep using `sat` instead of introducing new names like `satellite` and `satellite1`, and you should be able to see the routine’s effect.
